### PR TITLE
Pull Request for Issue1517: Generic scan config view not showing correct start and end values

### PR DESCRIPTION
**Pull request by iainworkman**
Added a check in the constructor of AMGenericStepScanConfigurationView which checks if a valid value already exists for each axis, and only sets defaults if they don't

Estimate: 1

### CODE REVIEW

**Comment by dretrex** [comment]
Don't forget to provide an estimate and assign it to someone.

**Comment by iainworkman** [comment]
I tried using a curl script that David provided me for setting the estimates, but it complained about certificates. It's about a 1. I'll try and remember to set it once I get home tonight.

**Comment by dretrex** [comment]
I set it for you, no worries.

File: source/ui/acquaman/AMGenericStepScanConfigurationView.cpp [hunk 1]
```diff
@@ -106,8 +106,15 @@ AMGenericStepScanConfigurationView::AMGenericStepScanConfigurationView(AMGeneric
 
 	positionsBox->setLayout(positionLayout);
 
-	onAxisControlChoice1Changed();
-	onAxisControlChoice2Changed();
+	// Set initial spinbox values in cases where we don't yet have axes:
+	if(configuration_->scanAxes().count() == 0) {
```
Comment (dretrex) [inline source/ui/acquaman/AMGenericStepScanConfigurationView.cpp]: My only question with this is whether or not it properly displays and enables everything.  If someone had a 2D scan would both axes show up enabled?

**Comment by dretrex** [comment]
Aside from the one comment it looks good.
Time estimate: 5 minutes

File: source/ui/acquaman/AMGenericStepScanConfigurationView.cpp (hunk 1 above)
Comment (iainworkman) [inline source/ui/acquaman/AMGenericStepScanConfigurationView.cpp]: I've tested the same change on the SGMUpgrade branch, and it behaves correctly with new scans with no controls yet selected, loaded 1d scan configurations and loaded 2d scan configurations.

Comment (dretrex) [inline source/ui/acquaman/AMGenericStepScanConfigurationView.cpp]: Okay, great.

**Pull request merged**